packages/router/src/__tests__/history.test.tsx

```diff
@@ -0,0 +1,19 @@
+import { navigate } from '../history'
+
+describe('navigate', () => {
+  it('does not add to the history if they are already at that location', () => {
+    expect(global.history.length).toEqual(1)
+    navigate('/test-1')
+    expect(global.history.length).toEqual(2)
+    navigate('/test-1')
+    expect(global.history.length).toEqual(2)
+  })
+
+  it('does add to the history if they are already at that location', () => {
+    expect(global.history.length).toEqual(2)
+    navigate('/test-2')
+    expect(global.history.length).toEqual(3)
+    navigate('/test-3')
+    expect(global.history.length).toEqual(4)
+  })
```

Comment:
Thoughts on this test. I was having a hard time figuring out how to clear the global history. Open to changes if necessary

Reply:
I know history is readonly in the browser, but I wonder if that's true for jest.



I don't think it's a huge problem, since I think history is reset for every test file.

Reply:
Yeah I searched around for a bit couldn't find a way. Tried something like
``` global.history = new History()```
 but typescript yells that it is an illegal constructor.  Dunno if there is a way to reset the environment between the tests And I believe it does reset the history after every test file though

Reply:
🆒 🆒 🆒 , I actually believe I read the same, but I couldn't find it in the docs. Maybe the best thing to do is leave a comment that we believe the history is reset in each test file, but couldn't find docs for it.